barretenberg/cpp/src/barretenberg/vm2/tracegen/execution_trace.cpp

```diff
@@ -779,6 +798,30 @@ void ExecutionTraceBuilder::process_registers(ExecutionOpCode exec_opcode,
         trace.set(REGISTER_MEM_TAG_COLUMNS[i], row, static_cast<uint8_t>(registers[i].get_tag()));
     }
 }
+void ExecutionTraceBuilder::process_get_env_var_opcode(
+    TaggedValue envvar_enum, TaggedValue output, TraceContainer& trace, uint32_t row, bool no_error)
+{
+    assert(envvar_enum.get_tag() == ValueTag::U8);
+    const auto& envvar_spec = GetEnvVarSpec::get_table(envvar_enum.as<uint8_t>());
```

Comment:
Are you sure `envvar_enum` will always be an U8? otherwise this will crash.

Reply:
Ah yes, you know from instr fetching :) you only need to do the conversion because you are getting it from the resolved operands, if this was a gadget you'd get it as u8 in the event.